src/commands/post.commands.ts

```diff
@@ -116,34 +117,73 @@ export function getPostCategory(categoryId: number) {
 
 export function createPost(post: any) {
     post.authorId = post.author.id;
-    post.postCategoryId = post.category.id;
-    return Post.create(post);
-}
-
-export function updatePost(post: any): Promise<IPostInstance> {
-    post.authorId = post.author.id;
     post.categoryId = post.category.id;
     let topics: any[] = post.topics;
-    let resultPost: IPostInstance;
     return database.transaction(function(t) {
-        return Post.update(post, { where: { id: post.id }, transaction: t }).then(() => {
-            // Update post
-            return Post.findOne({ where: { id: post.id }, transaction: t });
-        }).then(p => {
+        return Post.create(post, { transaction: t }).then(p => {
             // Update topics
-            resultPost = p;
             if (topics && topics.length) {
                 let queue = [];
                 topics.forEach(topic => {
-                    queue.push(Topic.findOne({ where: { id: topic.id } }));
+                    queue.push(Topic.findById(topic.id, { transaction: t }));
                 });
 
                 return Promise.all(queue).then(topicsArray => {
-                    return p.setTopics(topicsArray, { transaction: t });
+                    return p.setTopics(topicsArray, { transaction: t }).then(() => p);
                 });
             } else {
-                return p.setTopics([], { transaction: t });
+                return p.setTopics([], { transaction: t }).then(() => p);
             }
         });
+    }).then(post => {
+        let postJson: IPostAttributes = post.toJSON();
+        if (postJson.publishedAt) {
+            triggerWebhooks(WebhookEvents.PublishPost, post.toJSON());
+        } else {
+            triggerWebhooks(WebhookEvents.CreatePost, post.toJSON());
+        }
+        return post;
+    }).catch((err) => {
+        console.error(err.stack || err);
+        throw new Error('Unable to create post');
+    });
+}
+
+export function updatePost(post: any) {
+    post.authorId = post.author.id;
+    post.categoryId = post.category.id;
+    let topics: any[] = post.topics;
+    let published = false;
+    return database.transaction(function(t) {
+        return Post.findById(post.id, { transaction: t }).then(p => {
```

Comment:
if the specific post.id is not found in the database, is an exception thrown (and handled in the catch block), or is there a promise return that can also be used to handle an invalid post.id?  I know this is mostly a first pass, where you may not need to know if the passed in post object's id doesn't match anything in the db yet, so mostly just curious as I am not familiar with everything the ORM api offers.